src/components/atoms/Timeline/index.tsx

```diff
@@ -0,0 +1,171 @@
+import { padStart } from "lodash";
+import { FC, memo, ReactElement, useMemo } from "react";
+
+import { styled } from "@reearth/theme";
+
+export interface Props {
+  /**
+   * @description
+   * This value need to be epoch time.
+   */
+  currentTime: number;
+  /**
+   * @description
+   * These value need to be epoch time.
+   */
+  timelineRange: [start: number, end: number];
+  /**
+   * @description
+   * This value need to be in seconds.
+   * Default value is 600sec(10min).
+   */
+  memoryInterval?: number;
+  renderIcon: () => ReactElement;
+}
+
+const EPOCH_SEC = 1000;
+const PADDING_HORIZONTAL = 10;
+const STRONG_MEMORY_HOURS = 3;
+const STRONG_MEMORY_WIDTH = 2;
+const NORMAL_MEMORY_WIDTH = 1;
+const GAP_HORIZONTAL = 7;
+const HOURS_SECS = 3600;
```

Comment:
Do you think these will ever be customizable? If not, you can ignore this comment.

Reply:
These are not customizable by user but if design is changed then we can change these value.